code/android-target-library/src/main/java/com/adobe/marketing/mobile/target/TargetRequestBuilder.java

```diff
@@ -561,21 +561,20 @@ private JSONObject getExperienceCloudObject(final String visitorBlob, final Stri
 	 * @param customerIDs the {@code List<VisitorID>} of customer visitor id
 	 * @return the {@code JSONArray} generated
 	 */
-	private JSONArray getCustomerIDs(final List<VisitorID> customerIDs) {
+	private JSONArray getCustomerIDs(final List<Map<String, Object>> customerIDs) {
 		final JSONArray customerIDsArrayNode = new JSONArray();
 
 		try {
-			for (VisitorID visitorID : customerIDs) {
-				JSONObject newVisitorIDNode = new JSONObject();
-				newVisitorIDNode.put(TargetJson.CustomerIds.ID, visitorID.getId());
-				newVisitorIDNode.put(TargetJson.CustomerIds.INTEGRATION_CODE, visitorID.getIdType());
-				newVisitorIDNode.put(TargetJson.CustomerIds.AUTHENTICATION_STATE,
-						getAuthenticationStateToString(visitorID.getAuthenticationState()));
+			for (Map<String, Object> visitorID : customerIDs) {
+				final JSONObject newVisitorIDNode = new JSONObject();
+				newVisitorIDNode.put(TargetJson.CustomerIds.ID, String.valueOf(visitorID.get(TargetConstants.Identity.VISITOR_IDS_KEY_ID)));
```

Comment:
Can we use `DataReader.getString()` here? Same for line 571-573

Reply:
Core's internal serialization logic uses the same way : https://github.com/adobe/aepsdk-core-android/blob/7df5e898de5ecaf5b522ea27cf57b33384567eed/code/core/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java#L71

I followed the same thinking this might give bit more flexible to String cast other primitive datatype